pkg/plugins/runner.go

```diff
@@ -0,0 +1,81 @@
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	"get.porter.sh/porter/pkg/config"
+	"get.porter.sh/porter/pkg/context"
+	"github.com/pkg/errors"
+)
+
+type CommandOptions struct {
+	Command string
+}
+
+type PluginRunner struct {
+	*context.Context
+	pluginName string
+}
+
+func NewRunner(pluginName string) *PluginRunner {
+	return &PluginRunner{
+		Context:    context.New(),
+		pluginName: pluginName,
+	}
+}
+
+func (r *PluginRunner) Validate() error {
+	if r.pluginName == "" {
+		return errors.New("Plugin not specified")
+	}
+
+	pluginPath, err := config.New().GetPluginPath(r.pluginName)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get plugin path for %s", r.pluginName)
+	}
+
+	exists, err := r.FileSystem.Exists(pluginPath)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to stat path %s", pluginPath)
+	}
+	if !exists {
+		return errors.Errorf("Plugin %s doesn't exist in filesystem with path %s", r.pluginName, pluginPath)
+	}
+
+	return nil
+}
+
+func (r *PluginRunner) Run(commandOpts CommandOptions) error {
+	if r.Debug {
+		fmt.Fprintln(r.Err, "DEBUG Plugin Name: ", r.pluginName)
+		fmt.Fprintln(r.Err, "DEBUG Plugin Command: ", commandOpts.Command)
```

Comment:
I dont understand why we are using `r.Out` instead `r.Err` for debugging and why writing anything to `r.Out` not working. can you please explain ?

Reply:
Some of our commands takes stdout from the command's execution and parses it as a document. This is mainly (entirely?) due to the `schema` command which has the plugin output a json schema document to stdout which porter then parses and uses. If debug information was printed by either the mixin or porter, the document would be malformed. By writing debug lines to stderr, it helps us keep the two separate. 

I tried to document this a bit in our contributing guide under [Logging](https://github.com/deislabs/porter/blob/master/CONTRIBUTING.md#logging). If it's not clear, would you please let me know so that I can improve it?

Reply:
Got it. Actually i haven't read doc